dockertest/images.py

```diff
@@ -54,11 +54,11 @@ class DockerImage(object):  # pylint: disable=R0902
     #: parsing, spec defined in docker-io documentation.  e.g.
     #: ``[registry_hostname[:port]/][user_name/]
     #: (repository_name[:version_tag])``
-    repo_split_p = re.compile(r"(?P<repo_addr>.+?"  # optional
-                              r"(?P<repo_addr_port>:[\d]+?)?/)?"  # optional
-                              r"(?P<user>[\w\-\.\+]+/)?"  # optional
-                              r"(?P<repo>[\w\-\.]+)"  # required
-                              r"(?P<tag>:[\w\-\.]+)?")  # optional
+    repo_split_p = (r"(?P<repo_addr>.+?"  # optional
+                    r"(?P<repo_addr_port>:[\d]+?)?/)?"  # optional
+                    r"(?P<user>[\w\-\.\+]+/)?"  # optional
+                    r"(?P<repo>[\w\-\.]+)"  # required
+                    r"(?P<tag>:[\w\-\.]*)?")  # optional
```

Comment:
I don't quite follow, why can't events just reference it directly?  You can access the pattern in a compiled object by it's ``pattern`` attribute.  However, if the pattern needs changing to be used in events, then it's probably easier to just have a separate pattern that's more purpose specific (i.e. fewer groups or simpler).  I think that the regex in images is used so much, having a less precise pattern in the test is perfectly fine.